### ISSUE TITLE
Porting guide documentation error

### ISSUE THREAD

**Issue by tim-nordell-nimbelink** [aws/aws-iot-device-sdk-embedded-C#36]
For https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/master/PortingGuide.md, it states:
```c
IoT_Error_t iot_tls_is_connected(Network *pNetwork); Check if the TLS layer is still connected
```
However, the documentation inside network_interface.h states:
```c
IoT_Error_t (*isConnected)(Network *);    ///< Function pointer pointing to the network function to check if physical layer is connected
```
The usage is against the latter definition (e.g. your Ethernet cord is still plugged in or not) versus the former definition (e.g. your TLS layer got disconnected) in the only callee from within _aws_iot_mqtt_handle_reconnect(...).  It will not attempt a reconnect unless this function returns NETWORK_PHYSICAL_LAYER_CONNECTED which indicates that it's not checking if the TLS layer is still connected.

**Comment by chaurah** [comment]
Hi Tim,
       Thank you for pointing this out. The intention of the function, as you correctly state, is to determine if the Physical layer is connected, not the TLS layer. We should make that clearer and probably provide a separate API for checking the state of the TLS layer. I will add a fix for this to the next release. If you have any further thoughts on how we can handle this better please do let us know.

Rahul

**Comment by chaurah** [comment]
Fixed in v2.1.1 of the SDK. Closing.

**Comment by tim-nordell-nimbelink** [comment]
Hi!

While the newest changeset makes iot_tls_is_connect(...) consistent with isConnected(...), the changeset should have modified the documentation around iot_tls_is_Connect(...) and not isConnected(...).

- Tim
